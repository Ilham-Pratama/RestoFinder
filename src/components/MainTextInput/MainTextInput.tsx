import React from 'react';
import {
  StyleSheet,
  Text,
  TextInput,
  View,
  TextInputProps
} from 'react-native';
import { SvgIconProps } from 'assets/svgs/types';

interface IMainTextInput extends TextInputProps {
  icon?: React.FC<SvgIconProps>;
  disabled?: boolean;
}

const MainTextInput = (props: IMainTextInput) => {
  const { icon: Icon } = props;
  return (
    <View style={styles.container}>
      {Icon && <Icon height={20} width={20} />}
      {props.disabled ? (
        <Text
          accessibilityRole="text"
          style={[
            styles.disabledText,
            { color: props.value ? 'black' : 'gray' }
          ]}>
          {props.value || props.placeholder || ''}
        </Text>
      ) : (
        <TextInput
          accessibilityRole="search"
          placeholderTextColor="gray"
          style={[styles.textStyle, { marginLeft: Icon ? 10 : 0 }]}
          {...props}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#fff',
    borderRadius: 30,
    marginHorizontal: 20,
    paddingHorizontal: 12,
    shadowColor: 'black',
    shadowOffset: {
      height: 5,
      width: 5
    },
    elevation: 3,
    shadowOpacity: 0.1
  },
  textStyle: {
    color: 'black',
    flex: 1
  },
  disabledText: {
    paddingRight: 10,
    marginLeft: 15,
    marginVertical: 15
  }
});

export default MainTextInput;
