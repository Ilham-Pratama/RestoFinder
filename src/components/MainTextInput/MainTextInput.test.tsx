import { fireEvent, render } from '@testing-library/react-native';
import SearchIcon from 'assets/svgs/SearchIcon';
import React from 'react';
import MainTextInput from '.';

describe('MainTextInput', () => {
  const text = 'test update text';

  it('renders basic MainTextInput', () => {
    const changeFn = jest.fn();
    const component = render(<MainTextInput onChangeText={changeFn} />);

    const textInput = component.getByRole('search');

    expect(component.toJSON()).toMatchSnapshot();

    expect(textInput.props).toHaveProperty('placeholderTextColor', 'gray');
    expect(textInput.props).toHaveProperty('style.1.marginLeft', 0);

    fireEvent.changeText(textInput, text);

    expect(changeFn).toHaveBeenCalledWith(text);
  });

  it('renders MainTextInput with icon', () => {
    const changeFn = jest.fn();
    const component = render(
      <MainTextInput onChangeText={changeFn} icon={SearchIcon} />
    );

    const textInput = component.getByRole('search');

    expect(component.toJSON()).toMatchSnapshot();

    expect(component.getByRole('image').props).toHaveProperty('height', 20);
    expect(component.getByRole('image').props).toHaveProperty('width', 20);

    expect(textInput.props).toHaveProperty('style.1.marginLeft', 10);

    fireEvent.changeText(textInput, text);

    expect(changeFn).toHaveBeenCalledWith(text);
  });

  it('renders disabled MainTextInput', () => {
    const component = render(<MainTextInput value={text} disabled />);

    const textComponent = component.getByRole('text');

    expect(component.toJSON()).toMatchSnapshot();

    expect(textComponent.props).toHaveProperty('style.1.color', 'black');
  });

  it('renders disabled MainTextInput without value', () => {
    const component = render(<MainTextInput disabled />);

    const textComponent = component.getByRole('text');

    expect(component.toJSON()).toMatchSnapshot();

    expect(textComponent.props).toHaveProperty('style.1.color', 'gray');
  });
});
