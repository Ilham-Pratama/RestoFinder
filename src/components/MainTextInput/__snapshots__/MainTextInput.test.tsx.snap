// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTextInput renders MainTextInput with icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 30,
      "elevation": 3,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "paddingHorizontal": 12,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.1,
    }
  }
>
  <RNSVGSvgView
    accessibilityRole="image"
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    fill="#000"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={512}
    vbWidth={512}
    width={20}
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <TextInput
    accessibilityRole="search"
    icon={[Function]}
    onChangeText={[MockFunction]}
    placeholderTextColor="gray"
    style={
      Array [
        Object {
          "color": "black",
          "flex": 1,
        },
        Object {
          "marginLeft": 10,
        },
      ]
    }
  />
</View>
`;

exports[`MainTextInput renders basic MainTextInput 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 30,
      "elevation": 3,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "paddingHorizontal": 12,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.1,
    }
  }
>
  <TextInput
    accessibilityRole="search"
    onChangeText={[MockFunction]}
    placeholderTextColor="gray"
    style={
      Array [
        Object {
          "color": "black",
          "flex": 1,
        },
        Object {
          "marginLeft": 0,
        },
      ]
    }
  />
</View>
`;

exports[`MainTextInput renders disabled MainTextInput 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 30,
      "elevation": 3,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "paddingHorizontal": 12,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginVertical": 15,
          "paddingRight": 10,
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    test update text
  </Text>
</View>
`;

exports[`MainTextInput renders disabled MainTextInput without value 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 30,
      "elevation": 3,
      "flexDirection": "row",
      "marginHorizontal": 20,
      "paddingHorizontal": 12,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.1,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginVertical": 15,
          "paddingRight": 10,
        },
        Object {
          "color": "gray",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
